src/operator/tensor/indexing_op.cu

```diff
@@ -103,13 +162,125 @@ void SparseEmbeddingOpForwardRspImpl<gpu>(const OpContext& ctx,
   }
 }
 
+inline void SparseEmbeddingOpBackwardDeterministicRspImpl(const OpContext& ctx,
```

Comment:
Are the deterministic/nondeterministic versions divergent for more than 50% of their code or can they be combined somewhat? Looks kind of hard to maintain.